### PR TITLE
feat: [SRM-15375]: SLO Creation, deletion and update should be atomic

### PR DESCRIPTION
**Pull request by karansaraswat19**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52554)

### CODE REVIEW

**Comment by karansaraswat19** [comment]
trigger uts

**Review by deepak-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52554)* status: 3 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`953-events-api/src/main/java/io/harness/eventsframework/EventsFrameworkConstants.java` line 151 at r1](https://reviewable.io/reviews/harness/harness-core/52554#-NdhozZ2C9fGJZkwXIZ9:-NdhozZ2C9fGJZkwXIZA:b-9o522i) ([raw file](https://github.com/harness/harness-core/blob/39d833a03e8db8686b292c30c2b6b278899ba4b1/953-events-api/src/main/java/io/harness/eventsframework/EventsFrameworkConstants.java#L151)):*
> ```Java
>   public static final String SRM_SLO_CRUD_LOCK = "srm_slo_crud_lock";
>   public static final int SRM_SLO_CRUD_LOCK_TIMEOUT = 6;
>   public static final int SRM_SLO_CRUD_LOCK_WAIT_TIMEOUT = 4;
> ```

Lets keep 5 seconds for both. That way the curd operation will go through after 5 seconds. Else there might be calls which might fail after 4 seconds.
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/ServiceLevelObjectiveV2ServiceImpl.java` line 271 at r1](https://reviewable.io/reviews/harness/harness-core/52554#-Ndhqrg90q9Ufh6anY9g:-NdhqrgA1fmAgz7Xjghe:bmyy7vn) ([raw file](https://github.com/harness/harness-core/blob/39d833a03e8db8686b292c30c2b6b278899ba4b1/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/ServiceLevelObjectiveV2ServiceImpl.java#L271)):*
> ```Java
>       ProjectParams projectParams, String identifier, ServiceLevelObjectiveV2DTO serviceLevelObjectiveDTO) {
>     String lockString = SRM_SLO_CRUD_LOCK + identifier;
>     try (AcquiredLock acquiredLock = persistentLocker.waitToAcquireLock(lockString,
> ```

should add for create too as create and update command can come together.

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/ServiceLevelObjectiveV2ServiceImpl.java [hunk 1]
```diff
@@ -219,130 +226,143 @@ public TimeGraphResponse getOnboardingGraph(CompositeServiceLevelObjectiveSpec c
...
-        checkIfSLOPresent(projectParams, serviceLevelObjectiveDTO.getIdentifier());
-    ServiceLevelObjectiveV2DTO existingServiceLevelObjective =
-        sloEntityToSLOResponse(serviceLevelObjective).getServiceLevelObjectiveV2DTO();
-    List<String> serviceLevelIndicators = Collections.emptyList();
-    if (serviceLevelObjectiveDTO.getType().equals(ServiceLevelObjectiveType.SIMPLE)) {
-      validateSimpleSLO(serviceLevelObjectiveDTO, projectParams);
-      SimpleServiceLevelObjective simpleServiceLevelObjective = (SimpleServiceLevelObjective) serviceLevelObjective;
-      SimpleServiceLevelObjectiveSpec simpleServiceLevelObjectiveSpec =
-          (SimpleServiceLevelObjectiveSpec) serviceLevelObjectiveDTO.getSpec();
+    String lockString = SRM_SLO_CRUD_LOCK + identifier;
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/servicelevelobjective/services/impl/ServiceLevelObjectiveV2ServiceImpl.java:279]: This is not unique. Either we should have project Params, or using slo uuid

**Review by Varsha-lalwani — CHANGES_REQUESTED** [review]

**Comment by karansaraswat19** [comment]
trigger uts

**Review by deepak-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52554)* status: :shipit: complete! ✓

**Review by Varsha-lalwani — APPROVED** [review]

**Pull request merged**